### ISSUE TITLE
[Bug]: Invisible menu bar on Mac with camera notch

### ISSUE THREAD

**Issue by gusapps** [electron/electron#38499]
### Preflight Checklist

- [X] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/main/CONTRIBUTING.md) for this project.
- [X] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://www.github.com/electron/electron/issues) for a bug report that matches the one I want to file, without success.

### Electron Version

25.0.0

### What operating system are you using?

macOS

### Operating System Version

13.3.1

### What arch are you using?

arm64 (including Apple Silicon)

### Expected Behavior

In an Electron app with `cursor: none;` and a custom cursor in React, the default macOS cursor should remain hidden near the top edge of a fullscreen window. Example video from an older Mac without the camera notch:

https://github.com/electron/electron/assets/29574724/b94ef3ff-7b19-4538-901a-7fddd2a434ff

### Actual Behavior

At ~40px from the top edge of a fullscreen window, the Mac cursor is shown.

It is expected that the Mac cursor reappears once outside WebContents, but here it appears before it even reaches the blacked-out macOS menu bar. Screen recording from a Mac with camera notch:

https://github.com/electron/electron/assets/29574724/bb757528-3f2a-45d2-ad60-91d0488569b1

The 40px distance can be arrived at by rendering a `border-top: 40px solid red;`.

https://github.com/electron/electron/assets/29574724/afa23a28-cc5e-4f76-b9bf-3eab91149803

This should mean that it is not the window's title bar causing the cursor to reappear, since that is closer to 30px tall.

### Testcase Gist URL

https://gist.github.com/gdbroman/9772e73f74196e54bbdd100f0beb0653

### Additional Information

Kiosk mode does not solve this.

We have our custom cursor rendered in a separate, overlayed BrowserWindow, and the following almost solves this issue:

```
mainWindow.setAlwaysOnTop(true, 'screen-saver', 0);
mouseWindow.setAlwaysOnTop(true, 'screen-saver', 1);
```

But after exiting fullscreen once and re-entering fullscreen, the issue comes back.

**Comment by gdbroman** [comment]
Absolutely, here is an Electron Fiddle Gist for reproducing the issue @codebytere, let me know if anything needs clarification: https://gist.github.com/gdbroman/9772e73f74196e54bbdd100f0beb0653

**Comment by rossbulat** [comment]
I am also experiencing this issue.
